### ISSUE TITLE
Mocking of non-interface beans

### ISSUE THREAD

**Issue by Phil-Ba** [micronaut-projects/micronaut-test#20]
Hello,

the _@MockBean_ annotation only works if an interface is being the injection point. If you try to mock a bean which is being injected directly, the test will fail to inject anything. If this is by design it would be nice if there was some documentation on this behaviour, especially since generally injection works even if you don't provide interfaces for you beans.

**Comment by brandonlamb** [comment]
Please excuse no tone in text - isn't this a good thing from a design perspective? I thought that was one of the use-cases for interfaces?

* Documentation / intent
* Multiple implementations
* Mockable / swappable

Maybe a question of dogmatic vs pragmatic? Age old debate probably...

**Comment by Phil-Ba** [comment]
It sure is. That's why I suggested it to be at least documented clearly somewhere, since this is a very opinionated topic and from micronaut's perspective might be how it should be working.

- But from my experience 80-90 percent of interfaces in a normal business application end up with only one implementation anyways.
- Mocking is nowadays normally done with frameworks like mockito which don't require you to have an interface in the first place.
- Lastly, and this is very opinionated, I'd personally say that one of the keys to spring success was to not force the usage of interfaces on you during a time where EJB 1.0 did.

**Comment by iimpulse** [comment]
I agree with Phil-Ba here. I'd love to see mocking of non interfaced beans. Sometimes for small projects creating interfaces doesn't make sense because of tautology. For many projects interfaces are a great language design pattern, but not one that should be forced on the users of a framework.

**Comment by graemerocher** [comment]
This fix I believe requires changes to Micronaut core hence why we are taking a bit of extra time to look at it. We may be able to fix it after Micronaut 1.1 is out. Thanks for the patience.

**Comment by jameskleeh** [comment]
Please try this with the latest version. We have tests with `@MockBean` being used on an implementation. I'll reopen if the issue persists
